tuf/client/client.go

```diff
@@ -439,18 +446,20 @@ func (c *Client) downloadTargets(role string) error {
 	return nil
 }
 
-func (c *Client) downloadSigned(role string, size int64, expectedSha256 []byte) ([]byte, *data.Signed, error) {
-	rolePath := utils.ConsistentName(role, expectedSha256)
+func (c *Client) downloadSigned(role string, size int64, expectedHashes data.Hashes) ([]byte, *data.Signed, error) {
+	// TODO:(hukeping) not sure about this, since we havn't do the sanity check of the expectedHashes yes.
```

Comment:
We didn't previously do the sanity check because it was possible that the hash was empty (and so we wouldn't append any hash to the rolePath).  It's expected of the caller to pass any valid hash it wants to retrieve.

We have to make sure that we keep consistent with which checksums we download, cc @endophage @cyli for any ideas/timeframe on when we should allow downloading sha512 addresses